test/e2e/karmadactl_test.go

```diff
@@ -1154,6 +1154,87 @@ var _ = ginkgo.Describe("Karmadactl options testing", func() {
+		})
+
+		// testTaintEffect is a reusable function for testing taints on a given effect.
+		testTaintEffect := func(effect corev1.TaintEffect) {
+			ginkgo.It(fmt.Sprintf("should handle %s taints correctly", effect), func() {
+				taint := fmt.Sprintf("%s=%s:%s", taintKey, taintValue, effect)
+				taintRemove := fmt.Sprintf("%s-", taint)
+
+				ginkgo.By(fmt.Sprintf("Applying %s taint to the cluster", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taint)
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s tainted", clusterName))).Should(gomega.BeTrue())
+				})
+
+				ginkgo.By(fmt.Sprintf("Reapplying %s taint without --overwrite flag", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taint)
+					_, err := cmd.ExecOrDie()
+					gomega.Expect(err).Should(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(err.Error(), fmt.Sprintf("cluster %s already has environment taint(s) with same effect(s) and --overwrite is false", clusterName))).Should(gomega.BeTrue())
+				})
+
+				ginkgo.By(fmt.Sprintf("Overwriting %s taint with --overwrite flag", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taint, "--overwrite")
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s modified", clusterName))).Should(gomega.BeTrue())
+				})
+
+				ginkgo.By(fmt.Sprintf("Removing %s taint from the cluster", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taintRemove)
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, "untainted")).Should(gomega.BeTrue())
+				})
+
+				ginkgo.By(fmt.Sprintf("Running taint command in dry-run mode for %s", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taint, "--dry-run")
+					output, err := cmd.ExecOrDie()
+					gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+					gomega.Expect(strings.Contains(output, fmt.Sprintf("cluster/%s tainted", clusterName))).Should(gomega.BeTrue())
+				})
+
+				ginkgo.By(fmt.Sprintf("Verifying that %s taint was not applied during dry-run", effect), func() {
+					cmd := framework.NewKarmadactlCommand(kubeconfig, karmadaContext, karmadactlPath, "", karmadactlTimeout, "taint", "clusters", clusterName, taintRemove)
```

Comment:
You can directly check whether the cluster's taints match the expected conditions.